### PR TITLE
Life list

### PR DESCRIPTION
**Pull request by prakhar-s**
Added a table for a list of all unique species on the stats view of observation list page

### CODE REVIEW

File: src/components/pages/user/show/user-info-tabs/observation/life-list/table.tsx [hunk 1]
```diff
@@ -16,7 +16,6 @@ const Table = styled.table`
 
 const LifeListTable = ({ data, loadMore, title, extraParams }) => {
   const { t } = useTranslation();
-
   return data.list.length > 0 ? (
```
Comment (harshzalavadiya) [inline src/components/pages/user/show/user-info-tabs/observation/life-list/table.tsx]: This is unnecessary change please revert it to keep diff minimum

**Review by harshzalavadiya — CHANGES_REQUESTED** [review]
@prakhar-s except some minimal renamings this looks good please make make this minor modifications so we can merge this

File: src/components/pages/observation/list/views/stats/use-unique-species.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        } else {
+          _draft.list.push(...Object.entries(data.aggregateStatsData.groupUniqueSpecies));
+          _draft.lifelistoffset = _draft.lifelistoffset + LIFE_LIST_LIMIT;
+        }
+      });
+    }
+    setter((_draft) => {
+      _draft.isLoading = false;
+    });
+  };
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/use-unique-species.tsx]: I think this might be even better using only single setter

```js
setter((_draft) => {
    if (success) {
      if (reset) {
        _draft.list = Object.entries(data.aggregateStatsData.groupUniqueSpecies);
        _draft.lifelistoffset = LIFE_LIST_LIMIT;
      } else {
        _draft.list.push(...Object.entries(data.aggregateStatsData.groupUniqueSpecies));
        _draft.lifelistoffset = _draft.lifelistoffset + LIFE_LIST_LIMIT;
      }
    }
    _draft.isLoading = false;
  });
```

Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/use-unique-species.tsx]: made the required change

**Review by harshzalavadiya — APPROVED** [review]
